OfflineAudioContext/explainer.md

```diff
@@ -0,0 +1,230 @@
+
+## Proposed Approach - Add `startRenderingStream()` function
+
+The preferred approach is adding a new method `startRenderingStream()` that yields buffers of interleaved audio samples in a Float32Array, or another format as outlined in Open Questions. In this scenario, the user can read chunks as they arrive and consume them for storage, transcoding via WebCodecs, sending to a server, etc.
+
+Usage example:
+
+```js
+const context = new OfflineAudioContext({ numberOfChannels: 2, length: 44100, sampleRate: 44100 });
+
+// Add some nodes to build a graph...
+
+if ("startRenderingStream" in context) {
+  const reader = context.startRenderingStream({ format: "f32-interleaved" }).getReader(); 
+  while (true) {
+    // get the next chunk of data from the stream
+    const result = await reader.read();
+
+    // the reader returns done = true when there are no more chunks to consume
+    if (result.done) {
+      break;
+    }
+
+    // result.value contains interleaved Float32Array values
+    const buffers = result.value;
+  }
+} else {
+  audioBuffer = await offlineAudioContext.startRendering();
+}
+```
+
+Proposed interface:
+
+```js
+dictionary OfflineAudioRenderingOptions {
+    required AudioFormat format = "f32";
+}
+
+partial interface OfflineAudioContext {
+    // Returns a stream that yields buffers of interleaved audio samples in Float32Array or whatever format is specified
+    Promise<ReadableStream> startRenderingStream(options?: OfflineAudioRenderingOptions);
+};
+```
+
+### Pros
+
+- The new capability is feature detectable because it is a new function. Compared to Alternative 1 which cannot be easily detected
+- Aligns well with other web streaming APIs, similar to [WebCodecs](https://streams.spec.whatwg.org/#readablestream)
+- Works with very large durations, no upper limit to WebAudio graph duration
+
+### Cons
+
+- None of note
+
+### Output format
+
+There is an open question of what data format `startRenderingStream()` should return. The options under consideration are `AudioBuffer`, `Float32Array` planar or `Float32Array` interleaved.
+
+#### `AudioBuffer`
+
+**Pros**
+
+- semantically closest to the `startRendering()` API
+
+**Cons**
+
+- does not allow developers to BYOB (bring your own buffer) and BYOB helps developers manage memory usage, so `AudioBuffer` removes a bit of control
+
+#### Planar Float32Array
+
+**Pros**
+
+- `f32-planar` also already exists in the WebAudio spec
+
+**Cons**
+
+- requires the output of `startStreamingRendering()` to return an array of `Float32Array` in planar format for each output channel
+- this leaves a question of what to do if only one channel is read by the consumer, i.e. what should happen to the other channel's data?
+
+#### Interleaved Float32Array
+
+**Pros**
+
+- allows for streaming a single stream of data, rather than one for each channel
+
+**Cons**
+
+- None of note
+
+#### Recommendation
+
+`f32` [TODO - explain why]
```

Comment:
Let's either fill in the TODO or delete this section.  Since the other two options above already exist in Web Audio, we may have an easier time pursuing those options and perhaps exploring additional formats in the future.

Reply:
It sounds like the main benefit of interleaved f32 is that it will enable BYOB readers?  

We may want to add that as a pro above in "Interleaved Float32Array".